### PR TITLE
[W09B01] Added tests 

### PR DESCRIPTION
**Pull request by HendrikHuebner**

### CODE REVIEW

File: w08h03/test/pgdp/datastructures/UnitTest.java [hunk 1]
```diff
@@ -39,7 +39,8 @@ <Type extends Comparable<Type>> void testArray(Type[] input) {
 	@Test
 	@DisplayName("should iterate over empty graph")
 	public void emptyTest() {
-		assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+		Exception e = assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+		assertEquals(e.getMessage(), "Trying to call next on an empty QuarternarySearchTreeIterator");
```
Comment (nilsreichardt) [inline w08h03/test/pgdp/datastructures/UnitTest.java:43]: You accidentally added the commits for W08H03 😅 If you need help in removing this from the PR, please let me know 👍

**Comment by nilsreichardt** [comment]
FYI: To fix the broken `validate-pr-title` check, you need to merge the `main` branch into your `branch` and then just trigger the check again by renaming the title

**Review by TheAleWu — CHANGES_REQUESTED** [review]

**Comment by TheAleWu** [comment]
Other than that LGTM

**Review by MaximilianAnzinger — APPROVED** [review]

**Comment by nilsreichardt** [comment]
Tag me when this is ready to merge

**Comment by Garfield100** [comment]
yeah this folder structure is wrong :')

**Pull request merged**